src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -100,8 +103,12 @@ class TaskKiller @Inject() (
       toKill = appTasks
     )
 
-    appTasks.keys.find(id => !instanceTracker.hasSpecInstancesSync(id))
-      .map(id => Future.failed(PathNotFoundException(id)))
-      .getOrElse(killTasks)
+    async {
+      val allInstances = await(instanceTracker.instancesBySpec()).instancesMap
+      val (_, nonExisting) = appTasks.keys.partition(allInstances.contains)
```

Comment:
If you don't need the existing ones, you don't need `partition`.
Please use `filter` or `find`.
Sorry if I moved you into that direction.